pkg/proc/native/proc_linux.go

```diff
@@ -442,14 +460,26 @@ func (dbp *Process) stop(trapthread *Thread) (err error) {
 	if dbp.exited {
 		return &proc.ErrProcessExited{Pid: dbp.Pid()}
 	}
+
+	for _, th := range dbp.threads {
+		th.os.setbp = false
+	}
+	trapthread.os.setbp = true
+
+	// check if any other thread simultaneously received a SIGTRAP
+	for {
+		th, _ := dbp.trapWaitInternal(-1, trapWaitNohang)
+		if th == nil {
```

Comment:
Wouldn't we want to set `th.os.setbp = true` in this case as well (when `th != nil`)?

Reply:
trapWaitInternal sets setbp appropriately when the signal is SIGTRAP, doing it here would set it for threads that received the SIGSTOP we sent.